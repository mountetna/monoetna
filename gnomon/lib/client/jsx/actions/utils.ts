export function makeActionObject<T extends string, P>(type: T, payload: P): { type: T } & P {
    return { type, ...payload };
}