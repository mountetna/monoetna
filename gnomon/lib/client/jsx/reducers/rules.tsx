/* This isn't used in Rule creationâ€”only reading
from Magma for Name creation.
This should be split into reducers per key,
and eventually used for creation/editing. */

import { Rule, Synonym, Token } from '../models';
import { ACTION_TYPE, ADD_RULES } from '../actions/rules';



interface RulesState {
    rules: Rule[]
    tokens: Token[]
    synonyms: Synonym[]
}

const initialState: RulesState = {
    rules: [],
    tokens: [],
    synonyms: [],
}


export function rulesReducer(state: RulesState = initialState, action: ACTION_TYPE): RulesState {
    switch (action.type) {
        case ADD_RULES:
            return {
                rules: [...state.rules, ...action.rules],
                tokens: [...state.tokens, ...action.tokens],
                synonyms: [...state.synonyms, ...action.synonyms],
            }
        default: {
            return state;
        }
    }
}