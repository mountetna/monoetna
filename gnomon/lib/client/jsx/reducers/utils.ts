import { defaultDict } from "../utils/object";



// TODO: type to enforce T[idPropName]: string
export function listToIdObject<T>(list: T[], idPropName: string): Record<string, T> {
    const idObject = {};
    list.forEach(item => {
        idObject[item[idPropName]] = item
    })

    return idObject;
}

export function listToIdListObject<T>(list: T[], idPropName: string): Record<string, T[]> {
    const idObject = defaultDict<string, T[]>(_ => []);

    list.forEach(item => idObject[item[idPropName]].push(item))

    return { ...idObject } // remove default value getter
}