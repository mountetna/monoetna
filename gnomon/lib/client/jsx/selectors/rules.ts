import { createSelector } from 'reselect'

import { Rule, Token } from "../models";
import { RulesState } from '../reducers/rules';


interface State {
    rules: RulesState
}


export const selectRules = (state: State): Record<string, Rule> => state.rules.rules

export const selectRuleByName = (name: string) => (state: State): Rule => state.rules.rules[name]

export const selectTokens = (state: State): Record<string, Token> => state.rules.tokens

export const selectTokenByName = (name: string) => (state: State): Token => state.rules.tokens[name]

// const selectShopItems = state => state.shop.items
// const selectTaxPercent = state => state.shop.taxPercent

// const selectSubtotal = createSelector(selectShopItems, items =>
//   items.reduce((subtotal, item) => subtotal + item.value, 0)
// )