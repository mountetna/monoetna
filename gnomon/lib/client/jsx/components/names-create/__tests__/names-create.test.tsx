import React from 'react';
import { rest } from 'msw'
import { setupServer } from 'msw/node'
import { fireEvent, waitFor, waitForElementToBeRemoved, screen } from '@testing-library/react';

import NamesCreate from '../names-create';
import { renderWithProviders } from '../../../utils/test';
import * as exampleRulesResponse from './static/example-rules-response.json';
import { MagmaRulesResponse } from '../../../utils/rules';


global.ResizeObserver = jest.fn().mockImplementation(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
}))

const magmaHost = 'http://magma.test'
const projectName = 'example'
const rulesResponse: MagmaRulesResponse = exampleRulesResponse

const server = setupServer(
    rest.get(`${magmaHost}/gnomon/${projectName}`, (req, res, ctx) => {
        return res(ctx.json(rulesResponse));
    }),
)

beforeAll(() => server.listen());
afterEach(() => {
    server.resetHandlers();
    jest.restoreAllMocks();
});
afterAll(() => server.close());

describe('NamesCreate', () => {
    it("fetches rules and let's you add names", async () => {

        const component = renderWithProviders(
            <NamesCreate project_name={projectName} />
        )

        await waitFor(() => {
            expect(document.querySelector(
                'button[aria-label="Add Names"]:not(.Mui-disabled)'
            )).toBeTruthy()
        })
        expect(component.asFragment()).toMatchSnapshot()

        const rule = 'biospecimen'

        // open menu; add a rule name
        fireEvent.click(screen.getByLabelText('Add Names'))
        await waitFor(() => {
            expect(screen.queryByText(rule)).toBeTruthy();
        })
        fireEvent.click(screen.getByText(rule))

        // // // wait for menu to close; wait for rule group
        await waitForElementToBeRemoved(() => document.getElementById('add-names-rules-menu'))
        await screen.findByText(rule)

        expect(component.asFragment()).toMatchSnapshot()
    })

    
})