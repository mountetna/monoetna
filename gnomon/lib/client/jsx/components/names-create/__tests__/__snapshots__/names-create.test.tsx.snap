// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamesCreate fetches rules and let's you add names 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-projectAndToolbarContainer-1"
  >
    <div
      class="MuiGrid-root makeStyles-header-7 undefined MuiGrid-container MuiGrid-item"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-link-8 MuiTypography-colorPrimary"
        href="/example"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          example
        </h5>
      </a>
    </div>
    <div
      class="makeStyles-container-9"
    >
      <div
        class="makeStyles-buttonsOuterContainer-10"
      >
        <div
          class="makeStyles-buttonsContainer-11"
        >
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="true"
              aria-label="Add Names"
              class="MuiButtonBase-root-43 MuiButton-root-16 MuiButton-contained-24 makeStyles-button-14 MuiButton-containedPrimary-25 MuiButton-disableElevation-27"
              tabindex="0"
              title="Add Names"
              type="button"
            >
              <span
                class="MuiButton-label-17"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-44"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="true"
              aria-label="Find and Filter"
              class="MuiButtonBase-root-347 MuiButton-root-320 MuiButton-contained-328 makeStyles-button-14 MuiButton-containedPrimary-329 MuiButton-disableElevation-331 Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label-321"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-348"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="true"
              aria-label="Copy and Replace"
              class="MuiButtonBase-root-126 MuiButton-root-99 MuiButton-contained-107 makeStyles-button-14 MuiButton-containedPrimary-108 MuiButton-disableElevation-110 Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label-100"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-127"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H8V4h12v12zm-7-2h2v-3h3V9h-3V6h-2v3h-3v2h3z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="false"
              aria-label="Delete"
              class="MuiButtonBase-root-384 MuiButton-root-357 MuiButton-contained-365 makeStyles-button-14 MuiButton-containedPrimary-366 MuiButton-disableElevation-368 Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label-358"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-385"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4h-3.5z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="false"
              aria-label="Create All"
              class="MuiButtonBase-root-421 MuiButton-root-394 MuiButton-contained-402 makeStyles-button-14 MuiButton-containedPrimary-403 MuiButton-disableElevation-405 Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label-395"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-422"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm2 16H5V5h11.17L19 7.83V19zm-7-7c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3zM6 6h9v4H6z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="false"
              aria-label="Grow"
              class="MuiButtonBase-root-262 MuiButton-root-235 MuiButton-contained-243 makeStyles-button-14 MuiButton-containedSecondary-245 MuiButton-disableElevation-246"
              tabindex="0"
              title="Grow"
              type="button"
            >
              <span
                class="MuiButton-label-236"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-263"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 5.83L15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-header-7 placeholder MuiGrid-container MuiGrid-item"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-link-8 MuiTypography-colorPrimary"
        href="/example"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          example
        </h5>
      </a>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-composerList-5 MuiGrid-container"
  />
</DocumentFragment>
`;

exports[`NamesCreate fetches rules and let's you add names 2`] = `
<DocumentFragment>
  <div
    class="makeStyles-projectAndToolbarContainer-1"
  >
    <div
      class="MuiGrid-root makeStyles-header-7 undefined MuiGrid-container MuiGrid-item"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-link-8 MuiTypography-colorPrimary"
        href="/example"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          example
        </h5>
      </a>
    </div>
    <div
      class="makeStyles-container-9"
    >
      <div
        class="makeStyles-buttonsOuterContainer-10"
      >
        <div
          class="makeStyles-buttonsContainer-11"
        >
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="true"
              aria-label="Add Names"
              class="MuiButtonBase-root-43 MuiButton-root-16 MuiButton-contained-24 makeStyles-button-14 MuiButton-containedPrimary-25 MuiButton-disableElevation-27"
              tabindex="0"
              title="Add Names"
              type="button"
            >
              <span
                class="MuiButton-label-17"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-44"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="true"
              aria-label="Find and Filter"
              class="MuiButtonBase-root-632 MuiButton-root-605 MuiButton-contained-613 makeStyles-button-14 MuiButton-containedPrimary-614 MuiButton-disableElevation-616"
              tabindex="0"
              title="Find and Filter"
              type="button"
            >
              <span
                class="MuiButton-label-606"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-633"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="true"
              aria-label="Copy and Replace"
              class="MuiButtonBase-root-126 MuiButton-root-99 MuiButton-contained-107 makeStyles-button-14 MuiButton-containedPrimary-108 MuiButton-disableElevation-110 Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label-100"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-127"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H8V4h12v12zm-7-2h2v-3h3V9h-3V6h-2v3h-3v2h3z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="false"
              aria-label="Delete"
              class="MuiButtonBase-root-669 MuiButton-root-642 MuiButton-contained-650 makeStyles-button-14 MuiButton-containedPrimary-651 MuiButton-disableElevation-653 Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label-643"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-670"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4h-3.5z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="false"
              aria-label="Create All"
              class="MuiButtonBase-root-706 MuiButton-root-679 MuiButton-contained-687 makeStyles-button-14 MuiButton-containedPrimary-688 MuiButton-disableElevation-690 Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label-680"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-707"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm2 16H5V5h11.17L19 7.83V19zm-7-7c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3zM6 6h9v4H6z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="makeStyles-container-13 "
          >
            <button
              aria-haspopup="false"
              aria-label="Grow"
              class="MuiButtonBase-root-262 MuiButton-root-235 MuiButton-contained-243 makeStyles-button-14 MuiButton-containedSecondary-245 MuiButton-disableElevation-246"
              tabindex="0"
              title="Grow"
              type="button"
            >
              <span
                class="MuiButton-label-236"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-263"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 5.83L15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-header-7 placeholder MuiGrid-container MuiGrid-item"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-link-8 MuiTypography-colorPrimary"
        href="/example"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          example
        </h5>
      </a>
    </div>
  </div>
  <div
    class="makeStyles-countsList-2"
  >
    <div
      class="makeStyles-container-553 makeStyles-readyCounts-3"
    >
      <div
        class="makeStyles-count-554 count count-total"
      >
        1 name
      </div>
      <div
        class="separator"
      >
        •
      </div>
      <div
        class="makeStyles-count-554 count count-not-ready"
      >
        1 not ready
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-composerList-5 MuiGrid-container"
  >
    <div
      class="MuiGrid-root makeStyles-composer-6 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-4"
    >
      <div
        class="makeStyles-gridContainer-556"
      >
        <div
          class="makeStyles-innerContainer-557"
        >
          <div
            class="MuiGrid-root makeStyles-headerContainer-558 MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="makeStyles-tools-559"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-567 MuiCheckbox-root MuiCheckbox-colorSecondary makeStyles-checkbox-560 MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      aria-label="Select the Name Group"
                      class="PrivateSwitchBase-input-570"
                      data-indeterminate="false"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                </span>
                <button
                  aria-label="Toggle Expand/Collapse"
                  class="MuiButtonBase-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.41 18.59L8.83 20 12 16.83 15.17 20l1.41-1.41L12 14l-4.59 4.59zm9.18-13.18L15.17 4 12 7.17 8.83 4 7.41 5.41 12 10l4.59-4.59z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Add Name"
                  class="MuiButtonBase-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Delete Names"
                  class="MuiButtonBase-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4h-3.5z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="makeStyles-ruleNameTitle-561"
              >
                biospecimen
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="makeStyles-container-553 makeStyles-readyCounts-562"
              >
                <div
                  class="makeStyles-count-554 count count-total"
                >
                  1 name
                </div>
                <div
                  class="makeStyles-count-554 count count-not-ready"
                >
                  1 not ready
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-composersContainer-563 some-not-ready "
          >
            <div
              class="placeholder"
            >
              ...
            </div>
            <div
              class="makeStyles-composerList-564"
            >
              <div
                class="makeStyles-composerRowContainer-565"
              >
                <div
                  class="makeStyles-composerContainer-566"
                >
                  <div
                    class="makeStyles-container-571 makeStyles-container-578  "
                  >
                    <span
                      class="makeStyles-toolsContainer-572"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-567 MuiCheckbox-root MuiCheckbox-colorSecondary makeStyles-checkbox-573 MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            aria-label="Select Name"
                            class="PrivateSwitchBase-input-570"
                            data-indeterminate="false"
                            type="checkbox"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                        </span>
                      </span>
                      <button
                        aria-label="Copy Name with Values\\""
                        class="MuiButtonBase-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
                          />
                        </svg>
                      </button>
                      <button
                        aria-label="Delete Name\\""
                        class="MuiButtonBase-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4h-3.5z"
                          />
                        </svg>
                      </button>
                    </span>
                    <span
                      class="makeStyles-editorsContainer-574"
                    >
                      <div
                        class="MuiFormGroup-root makeStyles-elementsEditorContainer-579 MuiFormGroup-row"
                      >
                        <div
                          class="makeStyles-selectValueContainer-580 "
                        >
                          <button
                            aria-haspopup="true"
                            aria-label="Select Value for project"
                            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-currentSelectValue-581  MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation Mui-disabled Mui-disabled"
                            disabled=""
                            tabindex="-1"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              EXAMPLE
                            </span>
                          </button>
                        </div>
                        <div
                          class="makeStyles-selectValueContainer-580 "
                        >
                          <button
                            aria-haspopup="true"
                            aria-label="Select Value for separator"
                            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-currentSelectValue-581  MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation Mui-disabled Mui-disabled"
                            disabled=""
                            tabindex="-1"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              -
                            </span>
                          </button>
                        </div>
                        <div
                          class="makeStyles-selectValueContainer-580 "
                        >
                          <button
                            aria-haspopup="true"
                            aria-label="Select Value for species"
                            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-currentSelectValue-581 makeStyles-unset-582 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              <span
                                class="MuiButton-startIcon MuiButton-iconSizeSmall"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                                  />
                                </svg>
                              </span>
                              SP
                            </span>
                          </button>
                        </div>
                        <span
                          class="makeStyles-ruleCounterField-584 makeStyles-unset-588"
                        >
                          <div
                            class="MuiFormControl-root makeStyles-form-585"
                          >
                            <button
                              aria-haspopup="false"
                              aria-label="Set Counter Value for \\"subject\\""
                              class="MuiButtonBase-root makeStyles-autoIncrementButton-586"
                              tabindex="0"
                              type="button"
                            >
                              +1
                            </button>
                            <div
                              class="MuiInputBase-root makeStyles-inputContainer-589 makeStyles-inputContainer-591 makeStyles-ruleCounterInput-587 MuiInputBase-formControl"
                              inputmode="numeric"
                              style="width: 0px;"
                            >
                              <input
                                aria-invalid="false"
                                class="MuiInputBase-input"
                                min="0"
                                placeholder="n"
                                type="number"
                                value=""
                              />
                            </div>
                            <div
                              class="makeStyles-ruler-590"
                            >
                              n
                            </div>
                          </div>
                        </span>
                      </div>
                      <div
                        class="MuiFormGroup-root makeStyles-elementsEditorContainer-579 MuiFormGroup-row"
                      >
                        <div
                          class="makeStyles-selectValueContainer-580 "
                        >
                          <button
                            aria-haspopup="true"
                            aria-label="Select Value for separator"
                            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-currentSelectValue-581  MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation Mui-disabled Mui-disabled"
                            disabled=""
                            tabindex="-1"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              -
                            </span>
                          </button>
                        </div>
                        <div
                          class="makeStyles-selectValueContainer-580 "
                        >
                          <button
                            aria-haspopup="true"
                            aria-label="Select Value for timepoint"
                            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-currentSelectValue-581  MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation Mui-disabled Mui-disabled"
                            disabled=""
                            tabindex="-1"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              E
                            </span>
                          </button>
                        </div>
                        <span
                          class="makeStyles-ruleCounterField-584 makeStyles-unset-588"
                        >
                          <div
                            class="MuiFormControl-root makeStyles-form-585"
                          >
                            <button
                              aria-haspopup="false"
                              aria-label="Set Counter Value for \\"timepoint\\""
                              class="MuiButtonBase-root makeStyles-autoIncrementButton-586"
                              tabindex="0"
                              type="button"
                            >
                              +1
                            </button>
                            <div
                              class="MuiInputBase-root makeStyles-inputContainer-589 makeStyles-inputContainer-592 makeStyles-ruleCounterInput-587 MuiInputBase-formControl"
                              inputmode="numeric"
                              style="width: 0px;"
                            >
                              <input
                                aria-invalid="false"
                                class="MuiInputBase-input"
                                min="0"
                                placeholder="n"
                                type="number"
                                value=""
                              />
                            </div>
                            <div
                              class="makeStyles-ruler-590"
                            >
                              n
                            </div>
                          </div>
                        </span>
                      </div>
                      <div
                        class="MuiFormGroup-root makeStyles-elementsEditorContainer-579 MuiFormGroup-row"
                      >
                        <div
                          class="makeStyles-selectValueContainer-580 "
                        >
                          <button
                            aria-haspopup="true"
                            aria-label="Select Value for biospecimen"
                            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-currentSelectValue-581 makeStyles-unset-582 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              <span
                                class="MuiButton-startIcon MuiButton-iconSizeSmall"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                                  />
                                </svg>
                              </span>
                              BSP
                            </span>
                          </button>
                        </div>
                        <span
                          class="makeStyles-ruleCounterField-584 makeStyles-unset-588"
                        >
                          <div
                            class="MuiFormControl-root makeStyles-form-585"
                          >
                            <button
                              aria-haspopup="false"
                              aria-label="Set Counter Value for \\"biospecimen\\""
                              class="MuiButtonBase-root makeStyles-autoIncrementButton-586"
                              tabindex="0"
                              type="button"
                            >
                              +1
                            </button>
                            <div
                              class="MuiInputBase-root makeStyles-inputContainer-589 makeStyles-inputContainer-593 makeStyles-ruleCounterInput-587 MuiInputBase-formControl"
                              inputmode="numeric"
                              style="width: 0px;"
                            >
                              <input
                                aria-invalid="false"
                                class="MuiInputBase-input"
                                min="0"
                                placeholder="n"
                                type="number"
                                value=""
                              />
                            </div>
                            <div
                              class="makeStyles-ruler-590"
                            >
                              n
                            </div>
                          </div>
                        </span>
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
